frontend/alert/components/AlertForm/index.tsx

```diff
@@ -180,17 +201,21 @@ const AlertForm = ({
             autoCompleteInputRef.current &&
             (autoCompleteInputRef.current.value === autofillSection || (autoCompleteInputRef.current.value !== "" && selectedCourses.size == 0))
         ) {
-            doAPIRequest("/api/alert/registrations/", "POST", {
-                section: autoCompleteInputRef.current.value,
-                auto_resubscribe: autoResub === "true",
+            isCourseOpen(autoCompleteInputRef.current.value).then(isOpen => {
+                if (!isOpen && autoCompleteInputRef.current) {
```

Comment:
This is minor, but maybe we could make a variable
```
const section = autoCompleteInputRef.current.value;
```
just below the if statement above, so we don't have to repeat this `autoCompleteInputRef.current` check. Also this would prevent a potential bug if the user changes the input field quickly after pressing submit (i.e. before the `isCourseOpen` request finishes).